src/Service.GraphQLBuilder/Directives/ForeignKeyDirectiveType.cs

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using HotChocolate.Types;
+
+namespace Azure.DataApiBuilder.Service.GraphQLBuilder.Directives
+{
+    public class ForeignKeyDirectiveType : DirectiveType
```

Comment:
In the scenario where relationship is not defined at the database layer but only in the config file, will the ForeginKeyDirective be added to those columns/fields specified in the config file?

If so, a generic name such as `RelationshipFieldDirective` seems more appropriate?

Reply:
The Foreign key directive is only added to fields which are referencing field. Only the referencing fields can be nullable. Not the referenced ones. Hence the directive name is kept as such. The presence of directive on referencing field indicates that we should make it optional. We cannot and should not add the directive to referenced fields. 

For the relationships present in config file, either source.fields or target.fields can act as referencing fields based on the request body. Hence the FK directive will be added to both of them.

Reply:
The name ForeignKeyDirective in seems to signify a directive type that is a result of a relationship defined in the database layer through FK constraints.  But that need not be true as the directive is added for cases where the relationship is configured only through the config file.

Hence, my take is that a more generic name seems appropriate:  `RelationshipFieldDirective` /`ReferencingFieldDirective`.